internals-js/src/definitions.ts

```diff
@@ -343,54 +343,39 @@ export interface Named {
 export type ExtendableElement = SchemaDefinition | NamedType;
 
 export class DirectiveTargetElement<T extends DirectiveTargetElement<T>> {
-  private _appliedDirectives: Directive<T>[] | undefined;
+  readonly appliedDirectives: Directive<T>[];
 
-  constructor(private readonly _schema: Schema) {}
+  constructor(
+    private readonly _schema: Schema,
+    directives?: readonly Directive<any>[],
+  ) {
+    this.appliedDirectives = directives?.map((d) => this.attachDirective(d)) ?? [];
+  }
 
   schema(): Schema {
     return this._schema;
   }
 
+  private attachDirective(directive: Directive<any>): Directive<T> {
+    // if the directive is not attached, we can assume we're fine just attaching it to use. Otherwise, we're "copying" it.
+    const toAdd = directive.isAttached()
+      ? new Directive(directive.name, directive.arguments())
```

Comment:
I've kept as-is because I think it's the only place we do this for now and I was lacking inspiration on naming (it's not fully cloning since it doesn't keep the parent in particular). We can extract into a method later if this is done in other places.